src/Platforms/AbstractPlatform.php

```diff
@@ -3882,6 +3884,46 @@ public function getTimeTypeDeclarationSQL(array $column)
         throw Exception::notSupported(__METHOD__);
     }
 
+    /**
+     * Builds DBMS specific SQL to be used to create time, datetime and datetimetz columns
+     *
+     * @param mixed[] $column       The column definition
+     * @param string  $decl         The base SQL for the column type e.g. TIMESTAMP or TIME
+     * @param string  $lengthIfZero The string to append to the base SQL where the length is specified as '0'
+     * @param string  $lengthIfNull The string to append to the base SQL where no length is specified
+     * @param string  $suffix       The SQL to add after the base SQL and length specifier e.g. 'WITH TIME ZONE'
+     * @param int     $maxPrecision The maximum precision allowed on the platform
+     */
+    protected function buildDateTimeTypeDeclarationSQL(
+        array $column,
+        string $decl,
+        string $lengthIfZero = '',
+        string $lengthIfNull = '',
+        string $suffix = '',
+        int $maxPrecision = 6
+    ): string {
+        if (! isset($column['length']) || ! is_numeric($column['length'])) {
+            return trim($decl . $lengthIfNull . ' ' . $suffix);
```

Comment:
Why does the result need to be trimmed?

Reply:
To remove the trailing space when $suffix is null (to ensure absolutely consistent with what is currently generated).